contracts/utils/StrikeSelection.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.3;
+pragma experimental ABIEncoderV2;
+
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import {IOtoken} from "../interfaces/GammaInterface.sol";
+import {DSMath} from "../lib/DSMath.sol";
+import {IOptionsPremiumPricer} from "../interfaces/IRibbon.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract StrikeSelection is DSMath, Ownable {
+    using SafeMath for uint256;
+
+    /**
+     * Immutables
+     */
+    IOptionsPremiumPricer public immutable optionsPremiumPricer;
+
+    // delta for options strike price selection (ex: 0.1 is 1)
+    uint256 public delta;
+    // step pct at which we will iterate over
+    // (ex: 500 = 50% means we will move in leaps of 50% of spot price)
+    uint256 public step;
+
+    event DeltaSet(uint256 oldDelta, uint256 newDelta, address owner);
+    event StepSet(uint256 oldStep, uint256 newStep, address owner);
+
+    constructor(address _optionsPremiumPricer) {
+        require(_optionsPremiumPricer != address(0), "!_optionsPremiumPricer");
+        optionsPremiumPricer = IOptionsPremiumPricer(_optionsPremiumPricer);
+        // set delta to 0.1
+        delta = 1;
+        // set step to 5%
+        step = 50;
+    }
+
+    /**
+     * @notice Gets the strike price satisfying the delta value
+     * given the expiry timestamp and whether option is call or put
+     * @param expiryTimestamp is the unix timestamp of expiration
+     * @param isPut is whether option is put or call
+     */
+
+    function getStrikePrice(uint256 expiryTimestamp, bool isPut)
+        external
+        returns (uint256 strikePrice)
+    {
+        // asset price
+        uint256 assetPrice = optionsPremiumPricer.getUnderlyingPrice();
+
+        // time to expiration in number of days
+        uint256 t = expiryTimestamp.sub(block.timestamp).div(1 days);
+
+        // For each asset prices with step of 'margin' (down if put, up if call)
+        //   if that assets getOptionDelta(currStrikePrice, t) == (isPut ? 1 - delta : delta) with certain margin of error
+        //        return strike price
+
+        bool pastDelta = false;
+        uint256 currStrike = assetPrice;
+        uint256 newDelta = isPut ? uint256(10).sub(delta) : delta;
+
+        while (!pastDelta) {
+            uint256 currDelta =
+                optionsPremiumPricer.getOptionDelta(currStrike, t);
+            if (
+                newDelta.sub(step.div(2).mul(newDelta)) <= currDelta &&
+                currDelta <= newDelta.add(step.div(2).mul(newDelta))
+            ) {
+                strikePrice = currStrike;
+            }
+            currStrike = isPut
+                ? currStrike.sub(currStrike.mul(step).div(1000))
+                : currStrike.add(currStrike.mul(step).div(1000));
```

Comment:
One desirable thing we should aim for is to select strike prices that have similar increments (+100 for ETH, +1000 for BTC) as Deribit, this allows people to easily hedge on Deribit

![image](https://user-images.githubusercontent.com/5748750/121358934-e738c980-c92a-11eb-9ce7-f472cc010b80.png)

The loop should iterate through the list with increments of +100: 3000, 3100, 3200, 3300

Reply:
so do you want to just make step an absolute term that can be manually changed? ex: setStep -> 100 for eth / setStep -> 1000. then if eth appreciates we might do setStep -> 200 for eth based on what deribit does

Reply:
Will leave that up to you, as long as we can minimize the iterations and gas use, I'm ok with either approach

Reply:
Yeah...think it makes sense to leave as is. Even if we do in absolute terms will still be imprecise (ex current price is 2374 and we have increments of 100 will be 2474, 2574 so wont match perfectly either way.) Can always modulate iterations by increasing step

Reply:
If we iterate to 2474, we can always do a modulo so that it resolves correctly to 2400

`strike = strike - (strike % 100)`